### DISCUSSION TITLE
A way to embed your rules in the customasm binary?

### DISCUSSION THREAD

**Discussion by drako0812** [hlorenzi/customasm#130] [Ideas]
A thought I has was: What if there was a way to embed the text of your cpu rules inside of the binary so that end users wouldn't have to know about including a file defining the cpu?

This could maybe be a part of the process of building customasm or something. I have a little knowledge of the way Rust and Cargo works, so maybe as a part of the "build script" if a certain extra file exists (ie. `cpu_def.rs`) it enables "embedded mode". And then `cpu_def.rs` includes a string variable that includes the contents of your cpu definition which gets automatically processed at the beginning of all customasm invocations.

**Comment by hlorenzi** [comment]
Oh right, so you're building a modified version of the executable? I was ready to give you some sample code, but then I realized that the abstraction layer I'm using for the file system doesn't allow for non-existent files right now... So that might need some work!

In case you're just using the official executable, you can pass multiple file names as arguments (e.g. `customasm cpu_def.asm user_code.asm`) so that the user doesn't need to remember to use `#include` manually inside their file.

**Reply by drako0812** [reply]
Yeah, I knew about that... I think my inexperience with writing shell/Batch scripts so that I could just have the user use that as the "assembler" was why I was looking for an alternative. I'm sure if I spent some time on it, I'd figure it out. I just thought this would be a neat alternative.
